game-core/src/main/java/games/strategy/engine/framework/startup/ui/editors/ForumPosterEditor.java

```diff
@@ -1,296 +0,0 @@
-            GridBagConstraints.NONE,
-            new Insets(0, 0, bottomSpace, labelSpace),
-            0,
-            0));
-    add(
-        forums,
-        new GridBagConstraints(
-            1,
-            row,
-            1,
-            1,
-            1.0,
-            0,
-            GridBagConstraints.EAST,
-            GridBagConstraints.HORIZONTAL,
-            new Insets(0, 0, bottomSpace, 0),
-            0,
-            0));
-    row++;
-    add(
-        topicIdLabel,
-        new GridBagConstraints(
-            0,
-            row,
-            1,
-            1,
-            0,
-            0,
-            GridBagConstraints.WEST,
-            GridBagConstraints.NONE,
-            new Insets(0, 0, bottomSpace, labelSpace),
-            0,
-            0));
-    add(
-        topicIdField,
-        new GridBagConstraints(
-            1,
-            row,
-            1,
-            1,
-            1.0,
-            0,
-            GridBagConstraints.EAST,
-            GridBagConstraints.HORIZONTAL,
-            new Insets(0, 0, bottomSpace, 0),
-            0,
-            0));
-    add(
-        viewPosts,
-        new GridBagConstraints(
-            2,
-            row,
-            1,
-            1,
-            0,
-            0,
-            GridBagConstraints.EAST,
-            GridBagConstraints.NONE,
-            new Insets(0, 2, bottomSpace, 0),
-            0,
-            0));
-    row++;
-    add(
-        new JLabel(""),
-        new GridBagConstraints(
-            0,
-            row,
-            1,
-            1,
-            0,
-            0,
-            GridBagConstraints.WEST,
-            GridBagConstraints.NONE,
-            new Insets(0, 0, bottomSpace, labelSpace),
-            0,
-            0));
-    row++;
-    add(
-        includeSaveGame,
-        new GridBagConstraints(
-            0,
-            row,
-            2,
-            1,
-            0,
-            0,
-            GridBagConstraints.WEST,
-            GridBagConstraints.NONE,
-            new Insets(0, 0, 0, 0),
-            0,
-            0));
-    add(
-        testForum,
-        new GridBagConstraints(
-            2,
-            row,
-            1,
-            1,
-            0,
-            0,
-            GridBagConstraints.EAST,
-            GridBagConstraints.NONE,
-            new Insets(0, 0, 0, 0),
-            0,
-            0));
-    row++;
-    add(
-        alsoPostAfterCombatMove,
-        new GridBagConstraints(
-            0,
-            row,
-            2,
-            1,
-            0,
-            0,
-            GridBagConstraints.WEST,
-            GridBagConstraints.NONE,
-            new Insets(0, 0, 0, 0),
-            0,
-            0));
-    setupListeners();
-  }
-
-  private void checkFieldsAndNotify() {
-    areFieldsValid();
-    readyCallback.run();
-  }
-
-  /** Configures the listeners for the gui components. */
-  private void setupListeners() {
```

Comment:
Listener setup is inlined to constructor to update viewmodel based on component state, otherwise all other methods in this class are moved to 'viewModel' and/or broken up further.